### PR TITLE
Add support for GDEH0154D67 (aka epd1in54_v2)

### PR DESCRIPTION
**Pull request by jcard0na**
The main changes from the previous version of the display, are the
lookup tables and the initialization procedures.  Also, partial vs. full
updates have a slightly different logic.

The changes were mostly ported from the vendor driver (EPAPER.c) as well
as Waveshare example driver (EPD_1in54_V2.cpp).  Both drivers are
released under public "use at will" license.

This was tested as:

```
use epd_waveshare::{epd1in54_v2::*, prelude::*};

let mut epd = Epd1in54::new(&mut spi, cs_pin, busy_in, dc, rst, &mut delay)?;

epd.set_lut(&mut spi, Some(RefreshLut::Full))?;
epd.clear_frame(&mut spi, &mut delay)?;
epd.display_frame(&mut spi, &mut delay)?;

// display cleared, yay!

let mut display = Display1in54::default();

let bmp: Bmp<BinaryColor> = Bmp::from_slice(light_data)?;
let image = Image::new(&bmp, Point::zero());
image.draw(&mut display)?;

epd.set_lut(&mut spi, Some(RefreshLut::Quick))?;
epd.update_frame(&mut spi, &display.buffer(), &mut delay)?;
epd.display_frame(&mut spi, &mut delay)?;

// display shows image
```

#102 #79

### CODE REVIEW

**Comment by jcard0na** [comment]
One to the maintainer regarding this patch:

The file `graphics.rs` in this PR is an unmodified copy of the one in `epd1in54`.  I would be inclined to just link to that one instead, to avoid code duplication and ensure that future fixes and additions are applied to both displays.  What do you think?

**Comment by caemor** [comment]
> Question to the maintainer regarding this patch:
> 
> The file `graphics.rs` in this PR is an unmodified copy of the one in `epd1in54`. I would be inclined to just link to that one instead, to avoid code duplication and ensure that future fixes and additions are applied to both displays. What do you think?

Yes this sounds good, if they behave the same way.

File: src/type_a/constants.rs
Comment (caemor) [inline src/type_a/constants.rs:65]: I think it would be better to move the constants next to the epd1in54 v2 since they seem to be specific for this display (at least I didn't found the same sequence in here)

File: src/lib.rs
Comment (caemor) [inline src/lib.rs:79]: A `cargo fmt` might be necessary here

File: src/epd1in54_v2/graphics.rs
Comment (caemor) [inline src/epd1in54_v2/graphics.rs:1]: Just use the v1 one if there is no difference.

**Review by caemor — CHANGES_REQUESTED** [review]
Thanks for the PR. 
It nearly looks ready to merge besides the move of the graphics file and the LUTs.

**Comment by jcard0na** [comment]
I believe I addressed all the change requests.  Let me know if there's anything else that should be addressed before merging.

**Comment by caemor** [comment]
I think the new constants/LUTs file might be missing

**Comment by jcard0na** [comment]
Duh!  Sorry about that.  Just added it.

**Review by caemor — APPROVED** [review]

**Comment by lf94** [comment]
:o!!! Awesome work! Thank you!

**Pull request merged**
